src/fn/edit-contacts.js

```diff
@@ -0,0 +1,169 @@
+  const db = pouch();
+  const docDirectoryPath = args.docDirectoryPath;
+  fs.mkdir(docDirectoryPath);
+  const saveJsonDoc = doc => fs.write(`${docDirectoryPath}/${doc._id}.doc.json`, toDocs.toSafeJson(doc) + '\n');
+
+  const csvDir = `${environment.pathToProject}/csv`;
+  if(!fs.exists(csvDir)) {
+    warn(`No csv directory found at ${csvDir}.`);
+    return Promise.resolve();
+  }
+
+  return fs.recurseFiles(csvDir)
+  .filter(name => name.endsWith('.csv'))
+  .reduce((promiseChain, csv) =>
+    promiseChain
+    .then(() => {
+      info('Processing CSV file:', csv, '…');
+
+
+      const nameParts = fs.path.basename(csv).split('.');
+      const prefix = nameParts[0];
+      info('Processing CSV file:', prefix, '…');
+
+      switch(prefix) {
+        case 'contact':  return downloadDocs(csv, getIDs(csv),db, args);
+        default: throw new Error(`Unrecognised CSV type ${prefix} for file ${csv}`);
+      }
+    })
+    .then(docs => addToModel(csv, docs)),
+    Promise.resolve())
+
+  .then(() => model.exclusions.forEach(toDocs.removeExcludedField))
+  .then(() => Promise.all(Object.values(model.docs).map(saveJsonDoc)));
+};
+
+const model = {
+  csvFiles: {},
+  docs: {},
+  exclusions: [],
+  users: []
+};
+
+const addToModel = (csvFile, docs) => {
+    csvFile = csvFile.match(/^(?:.*[\/\\])?csv[\/\\](.*)\.csv$/)[1]; // eslint-disable-line no-useless-escape
+    model.csvFiles[csvFile] = docs;
+    docs.forEach(doc => {
+      model.docs[doc._id] = doc;
+    });
+  };
+
+function getIDs(csv) {
+    const { rows, cols } = fs.readCsv(csv);
+    var index = cols.indexOf('uuid');
+    if (index === -1){
+     throw Error('missing "uuid" column.');
+   }
+   return rows.map((item) => item[index]);
+}
+
+function processContacts(contactType, csv, ids, contactDocs, args){
+  const { rows, cols } = fs.readCsv(csv);
+  var colNames = args.colNames;
+  var index = cols.indexOf('uuid');
+  var toIncludeIndex = [];
+  if (colNames.length === 0) {
+    warn(' No columns specified, the script will add all the columns in the CSV!');
+    colNames = cols;
+  } else {
+    var splitFileColumns = [];
+    var fileColumn;
+    for(fileColumn of cols) {
+      if (fileColumn !== null) {
+        splitFileColumns.push(fileColumn.split(':')[0]);
+      }      
+    }
+    const columnsValid = colNames.every(function(column) { return splitFileColumns.includes(column);});
+    if (!columnsValid){
+      throw Error('The column name(s) specified do not exist.');
+    }
+
+    colNames = cols.filter(e => colNames.includes(e.split(':')[0]));
+    colNames.forEach(function(e){
+      toIncludeIndex.push(cols.indexOf(e));
+    });
+
+  }
+  if (colNames.includes('uuid')) {
+    colNames.splice(colNames.indexOf('uuid'),1);
+  }
+  return rows
+  .map(r => processCsv(contactType, colNames, r, ids, index, toIncludeIndex, contactDocs));
+
+}
+
+function processCsv(docType, columnNames, rowItem, ids, index, toIncludeIndex, contactDocs) {
+ var row = [];
```

Comment:
As far as I can tell, row is just a filtered version of `rowItem`.
Since you're not worried about mutating it (since you're doing a `splice` lower), you can use the param itself without the additional assignment.

Reply:
yeah, this was an unnecessary assignment. The unnecessary assignment has now been removed.